### PR TITLE
test(sbb-title): implement visual regression tests

### PR DESCRIPTION
**Pull request by MarioCastigliano**
not sure if it's worth testing all viewport sizes

### CODE REVIEW

File: src/elements/title/title.visual.spec.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    it(
+      'negative',
+      visualDiffDefault.with(async (setup) => {
+        await setup.withFixture(
+          html` <div style="padding: 1rem; background-color: var(--sbb-color-charcoal);">
+            <sbb-title negative>
+              Data without insights are trivial, and insights without action are pointless
+            </sbb-title>
+          </div>`,
+        );
```
Comment (TomMenga) [inline src/elements/title/title.visual.spec.ts:31]:
```suggestion
        await setup.withFixture(
          html`
            <sbb-title negative>
              Data without insights are trivial, and insights without action are pointless
            </sbb-title>
          `, { backgroundColor: negative ? 'var(--sbb-color-charcoal)' : undefined}
        );
```

File: src/elements/title/title.visual.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import { describeViewports, visualDiffDefault } from '../core/testing/private.js';
+
+import '../title.js';
+
+describe(`sbb-title`, () => {
+  describeViewports({ viewports: ['zero', 'small', 'large', 'ultra'] }, () => {
+    for (const level of ['1', '2', '3', '4', '5', '6']) {
+      it(
+        `visual level ${level}`,
```
Comment (TomMenga) [inline src/elements/title/title.visual.spec.ts:11]:
```suggestion
        `level=${level}`,
```

File: src/elements/title/title.visual.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { html } from 'lit';
+
+import { describeViewports, visualDiffDefault } from '../core/testing/private.js';
+
+import '../title.js';
+
+describe(`sbb-title`, () => {
+  describeViewports({ viewports: ['zero', 'small', 'large', 'ultra'] }, () => {
```
Comment (TomMenga) [inline src/elements/title/title.visual.spec.ts:8]: I'm conflicted about whether we should test all the viewports :/
@jeripeierSBB what do you think?

**Review by TomMenga — CHANGES_REQUESTED** [review]
Good job :)
Some minor fixes

File: src/elements/title/title.visual.spec.ts (hunk 3 above)
Comment (jeripeierSBB) [reply src/elements/title/title.visual.spec.ts:8]: I think the best would be ['zero', 'small', 'medium', 'large', 'ultra'] as for all those values, some title sizes are toggled specifically

Comment (jeripeierSBB) [reply src/elements/title/title.visual.spec.ts:8]: or maybe just use every viewport size...

File: src/elements/title/title.visual.spec.ts (hunk 1 above)
Comment (jeripeierSBB) [reply src/elements/title/title.visual.spec.ts:31]: I think it could be combined into a describeEach with title levels and negative prop in one loop.

File: src/elements/title/title.visual.spec.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    for (const level of ['1', '2', '3', '4', '5', '6']) {
+      it(
+        `visual level ${level}`,
+        visualDiffDefault.with(async (setup) => {
+          await setup.withFixture(
+            html`<sbb-title level=${level}>
+              Data without insights are trivial, and insights without action are pointless
```
Comment (jeripeierSBB) [inline src/elements/title/title.visual.spec.ts:15]: I'd suggest (even when it was not like that in Chromatic) to add a <p> tag below the title with a short random text to cover also the margins which are part of the sbb-title.

Comment (MarioCastigliano) [reply src/elements/title/title.visual.spec.ts:15]: Good idea

File: src/elements/title/title.visual.spec.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { html } from 'lit';
+
+import { describeViewports, visualDiffDefault } from '../core/testing/private.js';
+
+import '../title.js';
+
+describe(`sbb-title`, () => {
+  describeViewports({ viewports: ['zero', 'small', 'medium', 'large', 'ultra'] }, () => {
+    for (const level of ['1', '2', '3', '4', '5', '6']) {
```
Comment (jeripeierSBB) [inline src/elements/title/title.visual.spec.ts:9]: what about using
```
const cases = {negative: [false, true], level: [1,2,3,4,5,6]}
```
and the describeEach?
With that you can share the template and backgroundColor can be toggled by negative prop.

**Review by jeripeierSBB — APPROVED** [review]
LGTM, one possible refactor

File: src/elements/title/title.visual.spec.ts (hunk 5 above)
Comment (MarioCastigliano) [reply src/elements/title/title.visual.spec.ts:9]: I just assumed we didn't need to test the negative variant with all the levels and wanted to keep a relatively small number of screenshots. But I could live either way, what do you think it's best?

Comment (jeripeierSBB) [reply src/elements/title/title.visual.spec.ts:9]: excuse me, I misread the closing tag ;) It's fine like you did! :)

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
please adapt also the snapshot.spec.ts file

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thank you
